src/main/java/nl/uu/cs/ape/domain/BioToolsAPI.java

```diff
@@ -105,14 +124,15 @@ public static JSONArray getToolsFromEDAMTopic(String topicName) throws IOExcepti
 	 * @throws IOException
 	 * @throws JSONException
 	 */
-	private static JSONArray fetchToolListFromBioTools(JSONArray toolListJson) throws JSONException, IOException {
+	public static JSONArray fetchToolListFromBioTools(JSONArray toolListJson) throws JSONException, IOException {
 		JSONArray bioToolAnnotations = new JSONArray();
 		for (int i = 0; i < toolListJson.length(); i++) {
 			String currTool = toolListJson.getString(i);
 			Request request = new Request.Builder().url("https://bio.tools/api/" + currTool + "?format=json").build();
 			try (Response response = client.newCall(request).execute()) {
-				if (!response.isSuccessful())
-					throw new IOException("Unexpected code when trying to fetch" + response);
+				if (!response.isSuccessful()) {
+					log.error("The tool " + currTool + " could not be fetched from bio.tools.");
```

Comment:
After using log.error: Does this function still throw an IOException somewhere else (recursively)? Otherwise, the throws declaration could be reduced to only JSONException.